### DISCUSSION TITLE
How to get projection profile of image using convert?

### DISCUSSION THREAD

**Discussion by MonteShaffer** [ImageMagick/ImageMagick#4146] [Digital Image Processing]
For an example, I copied from a dissertation, PDF pg 24:

https://uknowledge.uky.edu/cgi/viewcontent.cgi?article=1055&context=cs_etds

![projection-profile_2021-09-02_10-30-33](https://user-images.githubusercontent.com/6663327/131862505-a5381b68-8076-4d71-a660-bea1a4fb38b7.png)

I would like to use 'convert' to get the 'x' profile, the 'y' profile, or both the "x,y" profiles simultaneously.  A profile would be a pixel count of data in each pixel bucket, and we are in grayscale or black/white mode [0,1] or {0,1} respectively.

Any ideas?

**Comment by dlemstra** [comment]
You could use the relatively new option `-sort-pixels` for this:

```
magick -pointsize 80 label:"projection profile" -sort-pixels output.png
```

This output is flipped compared to you your example image. You could `-rotate` the image first and you can also use that to get the vertical version but that requires a before and after rotate. 

Idea for the team: Maybe we could also do a sort into another direction inside ImageMagick where `+sort-pixels` would do a vertical sort instead? Or maytbe even better to introduce a "sort direction" option for that.

**Reply by fmw42** [reply]
If you are trying to get the rotation angle of the text, then rotate the image at different angles and then do the -sort-pixels. Or if you just want to know the maximum, you can rotate the image at different angles and then average the image to one column or one row depending upon what you are trying to do. Then find the column or row with the largest average. Or if you want dark, then get the row or column with the minimum average.

**Comment by fmw42** [comment]
If on Unix-like system (linux, mac osx or windows 10 unix) you can try my script textdeskew at my website at http://www.fmwconcepts.com/imagemagick/index.php. It gets the rotation angle of the text using FFT processing.
